pandas/core/frame.py

```diff
@@ -2317,6 +2317,56 @@ def to_records(
 
         return np.rec.fromarrays(arrays, dtype={"names": names, "formats": formats})
 
+    def to_redis(
+        self,
+        redis_conn: redis.client.Redis,
+        alias: str = None,
+        if_exists: str = "Overwrite",
+    ):
+        """Pandas IO method to cache a Dataframe to Redis. This uses pyarrow to serialize the Dataframe and stores it as a string value in Redis.
+
+        *Note* Redis has a 512 MB limit on values, so If the serialized Dataframe is over 512MB, a ValueError will be raised.
+
+        If you are using if_exists="Append", the following logic is applied:
+            - If the existing DataFrame has columns that don't exist in the new DataFrame, we just append which will add values to the columns that exist in the new DataFrame.
+            - If the new DataFrame has columns that don't exist in the existing DataFrame, we throw an error. This is due to a desire to maintain integrity of what has already been cached.
+
+        This operation is registered with Pandas under the redis namespace once installed. See example usage below:
+
+        Parameters
+        ----------
+        :param redis_conn: Redis connection object created by the python-redis library
```

Comment:
i forget the style name, but we use the convention

```
redis_conn : redis.client.Redis
    Redis connection object created by the python-redis library
```

Reply:
resolved in commit 21d13c7746aa4d91e91593581b1d04a681003421